src/ray/protobuf/common.proto

```diff
@@ -477,6 +477,25 @@ message ViewData {
   repeated Measure measures = 2;
 }
 
+enum TaskStatus {
+  // We don't have a status for this task because we are not the owner or the
+  // task metadata has already been deleted.
+  NIL = 0;
+  // The task is waiting for its dependencies to be created.
+  WAITING_FOR_DEPENDENCIES = 1;
+  // All dependencies have been created and the task is scheduled to execute.
+  SCHEDULED = 2;
+  // The task finished successfully.
+  FINISHED = 3;
+  // The task finished previously but is now being scheduled again because its
+  // outputs need to be reconstructed. It is waiting for its dependencies to be
+  // created again.
+  RECONSTRUCTING_AND_WAITING_FOR_DEPENDENCIES = 4;
```

Comment:
Should there be a "reconstructing" flag instead of separate states?

Reply:
I wanted to avoid adding an extra field to the protobuf but I can do this if you think it's cleaner.